mipengine/node/monetdb_interface/views.py

```diff
@@ -14,17 +18,33 @@ def create_view(
     table_name: str,
     columns: List[str],
     filters: dict,
+    privacy_protection=False,
 ):
     filter_clause = ""
     if filters:
         filter_clause = f"WHERE {build_filter_clause(filters)}"
     columns_clause = ", ".join(columns)
 
-    MonetDB().execute(
-        f"""
+    view_creation_query = f"""
         CREATE VIEW {view_name}
         AS SELECT {columns_clause}
         FROM {table_name}
         {filter_clause}
         """
-    )
+
+    MonetDB().execute(view_creation_query)
+
+    if privacy_protection:
+        view_rows_query_result = MonetDB().execute_and_fetchall(
+            f"""
+            SELECT COUNT(*)
+            FROM {view_name}
+            """
+        )
+        view_rows_result_row = view_rows_query_result[0]
+        view_rows_count = view_rows_result_row[0]
+
+        if view_rows_count < PRIVACY_THRESHOLD:
+            raise PrivacyError(
+                f"The following view has less rows than the PRIVACY_THRESHOLD({PRIVACY_THRESHOLD}):  {view_creation_query}"
```

Comment:
shouldn't the view be dropped right away?

Reply:
True, when a error is raised, it should be the controller that initiates a clean up process across all nodes for all tables/views created for this experiment (so for a specific context_id). I ll make a ticket.